### ISSUE TITLE
NTI: Object<?,?> type is a dictionary or not?

### ISSUE THREAD

**Issue by vobruba-martin** [google/closure-compiler#1361]
1)

``` javascript
/**
 * @type {Object<?,?>}
 */
var foo = {
    bar: 'baz'
};
console.log(foo.bar);
```

> No warning

2)

``` javascript
/**
 * @type {Object<?,?>}
 */
var foo = {
    bar: 'baz'
};
/**
 * @param {Object<?,?>} foo
 */
var fn = function(foo) {
    console.log(foo.bar);
};
fn(foo);
```

> WARNING - Cannot do '.' access on a dict
>   console.log(foo.bar);
>               ^
> 
> 0 error(s), 1 warning(s), 90.9% typed

I would say that there should be no warning in both cases.

**Comment by dimvar** [comment]
Yeah, this looks like a bug. Thanks for the report.

**Comment by Dominator008** [comment]
Looks like now we give a warning in the first case, but not in the second one.

**Comment by dimvar** [comment]
FYI, we're not planning to support parameterized Object in NTI. (We want to avoid spurious warnings, but not typecheck it.) The recommended type for parameterized objects is IObject.
